docs/call-tree.md

```diff
@@ -0,0 +1,353 @@
+
+## Frames and stacks in JavaScript
+
+Imagine this simplified example of 3 samples of mixed C++, JavaScript Code (js), and JIT optimized JavaScript (JIT). The functions are all labeled as to their implementation.
+
+| Sample index | Sample's stack |
+| ------------ | ---------------|
+| 0 | `JS::RunScript [c++]  ➡  onLoad [js]  ➡  a [js]  ➡  b [js]` |
+| 1 | `JS::RunScript [c++]  ➡  onLoad [js]  ➡  js::jit::IonCannon [c++]  ➡  a [JIT]  ➡  b [JIT]` |
+| 2 | `JS::RunScript [c++]  ➡  onLoad [js]  ➡  js::jit::IonCannon [c++]  ➡  a [JIT]  ➡  b [JIT]` |
+
+This example produces the following frames:
+
+| Frame index | Function name | Implementation |
+| ----------- | ------------- | -------------- |
+| 0 | `JS::RunScript` | C++ |
+| 1 | `onLoad` | JavaScript |
+| 2 | `a` | JavaScript |
+| 3 | `b` | JavaScript |
+| 4 | `js::jit::IonCannon` | C++ |
+| 5 | `a` | JIT |
+| 6 | `b` | JIT |
+
+For completeness, here is the stack table. It would only contain the frame index, and a stack prefix index, but it can be filled out with a little bit more information:
+
+| Stack index | Frame index | Prefix | Frame's function | Prefix's function |
+| ----------- | ----------- | ------ | ---------------- | ----------------- |
+| 0 | 0 | null | `JS::RunScript` | null |
+| 1 | 1 | 0    | `onLoad` | `JS::RunScript` |
+| 2 | 2 | 1    | `a | `onLoad` |
+| 3 | 3 | 2    | `b | `a` |
+| 4 | 4 | 1    | `js::jit::IonCannon` | `onLoad` |
+| 5 | 5 | 4    | `a` | `js::jit::IonCannon` |
+| 6 | 6 | 5    | `b` | `a` |
+
+Now, taking the stacks and building a call tree produces the following:
+
+```
+                  JS::RunScript
+                       ↓
+                     onLoad
+                   ↙       ↘
+                  a        js::jit::IonCannon
+                  ↓             ↓
+                  b             a
+                                ↓
+                                b
+```
+
+This is of course the correct stack of what you would want to see. But since we are mixing languages together into the same stack system, it might be nice to view only JS functions. In order to do that we hide any C++ stacks, and assign them to the nearest JS stack. Our tables would be updated to look like the following.
```

Comment:
Let's remove the "of course". See https://css-tricks.com/words-avoid-educational-writing/